forumone/generator-web-starter
# Refactor with Drupal 9 Support

### Goals

* [x] Extract common PHP CMS functionality to a common base generator
* [x] Scaffold Drupal 8 projects using a template repository (forumone/drupal-project)
* [x] Scaffold Drupal 9 projects using a template repository (forumone/drupal-project)
* [x] Maintain upgrade path for existing projects

## Test Cases

These tests are executed on each test profile to identify issues as documented below.

### 1. Yeoman execution

1. Build and install the current generator locally
```bash
npm install -g yo
npm ci
npm link
gulp
```
2. Change into the test scenario directory
```bash
cd test/<scenario>
```
3. Ensure the `tests/<scenario>` directory is empty with the exception of the committed `.yo-rc.json` file
```bash
git clean -fd .
```
4. Execute the generator in this test scenario directory. Select pre-filled options for all prompts.
```bash
yo web-starter install
```

### 2. Stand up local installation with `f1 up`

1. Within the test directory, execute `f1 up` to build and start all containers locally.
2. Open the site at http://localhost:8080
3. Clean up the test with `f1 down`

### 3. Test release image

1. Update `docker-compose.yml` to build and run the release image
    1. Change `services.<cms>.build.target` from `dev` to `release`
    2. Comment out all `services.<cms>` volumes
2. Build and start all containers locally with `f1 up`
3. Open the site at http://localhost:8080
4. Clean up the test with `f1 down`
